### PR TITLE
Add docs for UAVCAN ESC

### PR DESCRIPTION
**Pull request by hamishwillee**
This adds docs for UAVCAN ESC, in particular Sapog - and linking to Orel20 and Kotleta20. It draws heavily on the Zubax guide for configuring Sapog and PX4.

### CODE REVIEW

File: en/peripherals/uavcan_escs.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+PX4 supports the [UAVCAN](https://uavcan.org/) bus for connecting peripherals, including ESCs, GPS modules, various types of sensors, etc.
+
+UAVCAN ESCs have a number of advantages over [PWM ESCs and Servos](../peripherals/pwm_escs_and_servo.md):
+- UAVCAN has been specifically designed to deliver robust and reliable connectivity over relatively large distances.
+  It enables safe use of ESCs on bigger vehicles and communication redundancy.
+- The bus is bi-directional, enabling health monitoring and diagnostics.
+- Wiring is less complicated as you can have a single bus for connecting all your ESCs and other UAVCAN peripherals.
+- Setup is easier as you configure ESC numbering by manually spinning each motor.
+
+PX4 can be used with UAVCAN ESC that run [Sapog](https://github.com/PX4/sapog#px4-sapog) firmware; an advanced open source sensorless PMSM/BLDC motor controller firmware designed for use in propulsion systems of electric unmanned vehicles.
```
Comment (pavel-kirienko) [inline en/peripherals/uavcan_escs.md]: This is misleading. PX4 can be used with any UAVCAN ESC, not only Sapog-based (e.g., VESC, Zubax Myxa, etc).

File: en/peripherals/uavcan_escs.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+> **Note** At time of writing PX4 supports UAVCAN v0 (not v1.0).
+
+
+## Purchase
+
+Commercially available ESCs running Sapog firmware include:
```
Comment (pavel-kirienko) [inline en/peripherals/uavcan_escs.md]: Should it not be stated that there are other UAVCAN ESC on the market?

File: en/peripherals/uavcan_escs.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+  ![Orel20 - Top](../../assets/peripherals/esc_uavcan_zubax_orel20/orel20_top.jpg)
+
+- [Holybro Kotleta20](https://shop.holybro.com/kotleta20_p1156.html)
+
+  ![Kotleta20 - Top](../../assets/peripherals/esc_uavcan_holybro_kotleta20/kotleta20_top.jpg)
+  ![Kotleta20 - Bottom](../../assets/peripherals/esc_uavcan_holybro_kotleta20/kotleta20_bottom.jpg)
+
+
+## Setup
```
Comment (pavel-kirienko) [inline en/peripherals/uavcan_escs.md]:
```suggestion
## Sapog-based ESC setup
```

File: en/peripherals/uavcan_escs.md (hunk 2 above)
Comment (jinger26) [reply en/peripherals/uavcan_escs.md]: @pavel-kirienko  where can we locate the most up-to-date list for market available UAVCAN ESCs?

Comment (pavel-kirienko) [reply en/peripherals/uavcan_escs.md]: This is the least incomplete list known to me: https://forum.uavcan.org/t/uavcan-esc-options/452/3?u=pavel.kirienko

Comment (hamishwillee) [reply en/peripherals/uavcan_escs.md]: @pavel-kirienko Thanks very much. I have merged this with enough modification to make it clear that there are other options and with a link to your list. 

I'd like to do more, but none of [the links](https://forum.uavcan.org/t/uavcan-esc-options/452/3?u=pavel.kirienko) state support for PX4 or indicate that how they are configured. So from my perspective they aren't really useable.

Can you help me with this? A few questions
1. Is anything in your list other than Zubax Orel 20 and Holybro Koleta20 based on Sapog (ie can be used with same instructions.
2. [Mitochondrik](https://zubax.com/products/mitochondrik) is a chip with UAVCAN interface that is used/integrated in the [Sadulli](https://shop.zubax.com/collections/integrated-drives/products/sadulli-integrated-drive-open-hardware-reference-design-for-mitochondrik?variant=27740841181283) drive. So basically Mitochondric is basically the ESC and is what defines the UAVCAN support for a class of drives (ie it is like Sapog support) - right?
   - How is this used with PX4? Is [connecting](http://docs.px4.io/master/en/peripherals/uavcan_escs.html#connecting) the same for all uavcan systems - e.g. duplicate of sapog docs?
   - How is it enabled on PX4?
   - How is motor enumeration done?
3. Same questions for [Myxa](https://zubax.com/products/myxa) - connection the same? How enable on PX4, how enumerate motors.
4. VESC - same questions. 
5. I can chase up OlliW and NXP directly, but if you have any overview guidance you can share that would be great.

**Comment by hamishwillee** [comment]
@olliw42 I've written a very sapog-centric guide to UAVCAN escs on PX4 [here](http://docs.px4.io/master/en/peripherals/uavcan_escs.html). It looks like your [UC4H ESC-Actuator Node](http://www.olliw.eu/2017/uavcan-for-hobbyists/#chapterescactuator) basically provides a UAVCAN interface to any PWM or Dshot ESC (nice!). 

Have you tried it with PX4? Is there any particular configuration required flight-controller side to get this to work?

**Comment by pavel-kirienko** [comment]
They don't state support for PX4 because they support UAVCAN which is a
common interface that abstracts any irrelevant specifics of whatever there
is on the other end of the link away. PX4 supports UAVCAN; transitively,
any ESC that supports UAVCAN supports PX4.

Every listed product is usable with PX4. For comparison, when you buy a USB
keyboard or a webcam, you don't need the vendor to spell out for you
whether your version of mac book or whatever hardware/OS you prefer is
supported because you know it supports USB.

It is crucial to ensure that people understand this concept. It's obviously
explained in the UAVCAN specification but few read it, turning to doc pages
like this one instead, so it is paramount that we don't inadvertently
mislead people into thinking that UAVCAN is just a bad clone of I2C.

1. No.

2. PX4 doesn't care what kind of ESC there is on the other end of the link,
barring optional features like automated enumeration. They all look the
same to the autopilot. For comparison, Skype client doesn't care what model
your microphone is, because there's a standard abstraction layer provided.

**Comment by hamishwillee** [comment]
@pavel-kirienko I understand the theory but ...
1. my USB "thingies" are always asking me to install additional drivers
2. If setup is really common, why would you have changed the docs from "setup" to "Sapog setup"?
3. The devil is in the detail. You say that automated enumeration is an optional feature - but how does someone know that? How do they do the enumeration if it is not automated?

Hence my questions - what is common and what is different.

Proposal:
1. I add a note making it clear that all UAVCAN is largely plug'n play
2. I move [connection section](http://docs.px4.io/master/en/peripherals/uavcan_escs.html#connecting) so that it is not associated with Sapog - ie it is "common to all UAVCAN ESC".
3. I move [Enabling the UAVCAN driver](http://docs.px4.io/master/en/peripherals/uavcan_escs.html#enabling-the-uavcan-driver) into common to all UAVCAN ESC setup too. 

That just leaves the section on enumeration of motors. How is that done if auto-enumeration is not supported for the firmware?

**Comment by pavel-kirienko** [comment]
> my USB "thingies" are always asking me to install additional drivers

USB is a convenient analogy for UAVCAN because you can really stretch it far. In this case, you could picture it as ESC being a standard class of UAVCAN devices (there is no concept of "standard class" in UAVCAN but it helps) that needs no driver, just like a USB keyboard or a mouse, unlike, say, more complex things such as printers.

> If setup is really common, why would you have changed the docs from "setup" to "Sapog setup"?

Because the article goes on to explain how to configure the ESC itself, so it's a mix of PX4-specific and Sapog-specific settings. We could split it into two: a generic article about the PX4 side of things and a Sapog-specific article about that particular sort of ESC. Other articles could be added later focusing on other ESCs such as VESC or Myxa; the PX4 side instructions would still apply as I explained in the previous message.

> The devil is in the detail. You say that automated enumeration is an optional feature - but how does someone know that? How do they do the enumeration if it is not automated?

They read the user documentation for their ESC which says whether auto-enumeration is available or not. If it's not, they read on their ESC-specific instruction to find out how to assign ESC indexes manually.

> Proposal:

Seeing as you might be interested to expand the docs to accommodate other ESCs later on, would you consider splitting the article as I just suggested here above?

**Comment by hamishwillee** [comment]
@pavel-kirienko Thanks - yes, I think your proposal is the same as mine. For your review:  https://github.com/PX4/px4_user_guide/pull/552

**Pull request merged**